### ISSUE TITLE
bin File does not work at all

### ISSUE THREAD

**Issue by FOSV** [unreality/FujiHK#7]
Hi, I flashed the .bin via NodeMCU PyFlasher, the procedure seems to have been successful.

I disconnect and reconnect the Atom but when I try to run the procedure to start the SoftAP for pairing, the led doesn't flash and I don't see any AP to connect to.

So I thought I'd see from the serial monitor what was going on and that's what the Atom repeatedly spits out:

`rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:QIO, clock div:2
load:0x3fff0030,len:1184
load:0xddddcfcc,len:-808595987
1162 mmu set 00010000, pos 00010000
1162 mmu set 00020000, pos 00020000
1162 mmu set 00030000, pos 00030000
1162 mmu set 00040000, pos 00040000
1162 mmu set 00050000, pos 00050000
1162 mmu set 00060000, pos 00060000
1162 mmu set 00070000, pos 00070000
1162 mmu set 00080000, pos 00080000
1162 mmu set 00090000, pos 00090000
1162 mmu set 000a0000, pos 000a0000
1162 mmu set 000b0000, pos 000b0000
1162 mmu set 000c0000, pos 000c0000
1162 mmu set 000d0000, pos 000d0000
1162 mmu set 000e0000, pos 000e0000
1162 mmu set 000f0000, pos 000f0000
1162 mmu set 00100000, pos 00100000
1162 mmu set 00110000, pos 00110000
ets Jun  8 2016 00:22:57

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:QIO, clock div:2
load:0x3fff0030,len:1184
load:0xddddcfcc,len:-808595987
1162 mmu set 00010000, pos 00010000
1162 mmu set 00020000, pos 00020000
1162 mmu set 00030000, pos 00030000
1162 mmu set 00040000, pos 00040000
1162 mmu set 00050000, pos 00050000
1162 mmu set 00060000, pos 00060000
1162 mmu set 00070000, pos 00070000
1162 mmu set 00080000, pos 00080000
1162 mmu set 00090000, pos 00090000
1162 mmu set 000a0000, pos 000a0000
1162 mmu set 000b0000, pos 000b0000
1162 mmu set 000c0000, pos 000c0000
1162 mmu set 000d0000, pos 000d0000
1162 mmu set 000e0000, pos 000e0000
1162 mmu set 000f0000, pos 000f0000
1162 mmu set 00100000, pos 00100000
1162 mmu set 00110000, pos 00110000
ets Jun  8 2016 00:22:57`

**Comment by YouCanNotBeSerious** [comment]
In your debug info:



clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
**mode:QIO, clock div:2**
load:0x3fff0030,len:1184

But in the instructions:

Launch NodeMCU PyFlasher
Set your serial port
Select the downloaded .bin file as your NodeMCU firmware
**Select 'Dual I/O (DIO)' as Flash mode**
Erase flash to 'yes'
Click 'Flash NodeMCU'

Try DIO?

**Comment by FOSV** [comment]
Yes, I set the DIO mode in NodeMCU PyFlasher, I attach what the console says:

Command: esptool.py --port /dev/cu.usbserial-A95256C116 --baud 115200 --after no_reset write_flash --flash_size detect --flash_mode dio 0x00000 /Users/Francesco/Downloads/fuji-hk.bin --erase-all

esptool.py v3.0
Serial port /dev/cu.usbserial-A95256C116
Connecting....
Detecting chip type... ESP32
Chip is ESP32-PICO-D4 (revision 1)
Features: WiFi, BT, Dual Core, 240MHz, Embedded Flash, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 4c:75:25:d5:b1:e4
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Auto-detected Flash size: 4MB
Erasing flash (this may take a while)...
Chip erase completed successfully in 3.4s
Compressed 1068352 bytes to 671112...
Wrote 1068352 bytes (671112 compressed) at 0x00000000 in 60.4 seconds (effective 141.5 kbit/s)...
Hash of data verified.

Leaving...
Staying in bootloader.

Firmware successfully flashed. Unplug/replug or reset device 
to switch back to normal boot mode.

**Comment by HamidKetabdar** [comment]
Hi FOSV, I've encountered the same problem. Have you managed to find a solution for it?

**Comment by FOSV** [comment]
Unfortunately not, I no longer have time for resolution attempts and I think this repository has been abandoned.
I suggest you have a look on the HomeAssistant community, there is someone who has coded an integration that seems to work well.
Here is the link:
[https://community.home-assistant.io/t/fujitsu-ac-heat-pump-integration-via-esphome-esp32/407610](url)

**Comment by unreality** [comment]
Yeah, the .bin was provided as convinience but the best way forward is to compile it yourself and load it that way - you'd also get any bugfixes from upstream libraries as well.

**Comment by pauleec** [comment]
Its a pity, I've had exactly the same problem described above. Can anyone walk me through how to compile it myself and get the  esp32 M5Atoms I have purchased (I've bought 3 of the M5Atom and 1 M5Atom S3 and another different/cheaper ESP32 board I wanted to test recompiling for). This is an awesome project. It would be great to keep it alive and help the community looking to use it get it working with their Fuj head units. Pleaae respond to this post if you're interested and willing to help identify the issue, recompile and get it working properly. thanks!

**Comment by pauleec** [comment]
After following the instructions and ensuring it was flashed with DIO mode as requested I' also seeing the same output when I connect a serial monitor via Arduino through the USB running at 115200 baud. It seems to be in some kind of a bootloop and is churning out the following output:

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 188777542, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:QIO, clock div:2
load:0x3fff0030,len:1184
load:0xddddcfcc,len:-808595987
1162 mmu set 00010000, pos 00010000
1162 mmu set 00020000, pos 00020000
1162 mmu set 00030000, pos 00030000
1162 mmu set 00040000, pos 00040000
1162 mmu set 00050000, pos 00050000
1162 mmu set 00060000, pos 00060000
1162 mmu set 00070000, pos 00070000
1162 mmu set 00080000, pos 00080000
1162 mmu set 00090000, pos 00090000
1162 mmu set 000a0000, pos 000a0000
1162 mmu set 000b0000, pos 000b0000
1162 mmu set 000c0000, pos 000c0000
1162 mmu set 000d0000, pos 000d0000
1162 mmu set 000e0000, pos 000e0000
1162 mmu set 000f0000, pos 000f0000
1162 mmu set 00100000, pos 00100000
ets Jun  8 2016 00:22:57

Keep in mind this is just the USB-C cable plugged in and powering the device through the USB C port. I've tried powering the unit through the 3.3v/gnd pins but it has no effect on the M5ATOM (the USB serial port is still churnIng out the same output at 115200). Does the M5ATOM need to be flashed at a lower baud rate? What is causing it to bootloop like this? I will keep trying to reflash at different speeds but it appears the .bin file is not working with the currently available M5ATOMs that are on sale. I'm happy to recompile if I knew how to but my experience with arduino and the other tools is limited (would be good to learn how to do this though as I would like to try to get your code running with other ESP32 boards like the m5ATOM S3 and the C3-Mini boards. If youre able to find the time or can point me to a person that would be willing to spare some time to help me overcome this issue I would be really grateful... but right now i'm stuck on the bootloop (like the other guy) and having spent all this money on the electronics to use your project work would be really grateful if I could find a way forward with something that will make it boot properly. If you could revisit this issue and help  identify where we are making a mistake I would be enormously grateful. Thanks!

**Comment by pauleec** [comment]
This is the output of the NodeMCU PyfFasher when I flash at 115200 Baud to the M5Atom:

Command: esptool.py --baud 115200 --after no_reset write_flash --flash_size detect --flash_mode dio 0x00000 C:\Users\Paul\Downloads\fuji-hk.bin --erase-all

esptool.py v4.8.1
Found 1 serial ports
Serial port COM4
Connecting.....
Detecting chip type... Unsupported detection protocol, switching and trying again...
Connecting.....
Detecting chip type... ESP32
Chip is ESP32-PICO-D4 (revision v1.1)
Features: WiFi, BT, Dual Core, 240MHz, Embedded Flash, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: f0:24:f9:bb:40:d0
Uploading stub...
Running stub...
Stub running...
Configuring flash size...
Auto-detected Flash size: 4MB
Erasing flash (this may take a while)...
Chip erase completed successfully in 2.4s
Compressed 1068352 bytes to 671112...
Wrote 1068352 bytes (671112 compressed) at 0x00000000 in 60.5 seconds (effective 141.2 kbit/s)...
Hash of data verified.

Leaving...
Staying in bootloader.

Firmware successfully flashed. Unplug/replug or reset device 
to switch back to normal boot mode.

**Comment by pauleec** [comment]
Ok - I have good news! I have successfully imported the C++ code into my arduino IDE. I needed to add the Homespan and FujitsuHeatPump libraries to my IDE and also needed to use the ESP board library version 3.1.3 (I had to downgrade to the older version in order for Arduino to compile the project without any errors. 3.2.0 generated errors.

So the solution is to NOT use the .bin and flash tool provided but instead install Arduino and setup your IDE for the M5Atom. Go into board manager and downgrade the esp32 board library from 3.2.0 to 3.1.3 and import the Homespan and FujitsuHeatPump libraries to your local IDE library folder and then compile the sketch. If you get no errors  (in my case when the compile completed I got an output on my console saying "Sketch uses 1438024 bytes (45%) of program storage space. Maximum is 3145728 bytes.
Global variables use 62388 bytes (19%) of dynamic memory, leaving 265292 bytes for local variables. Maximum is 327680 bytes." which meant it completed successfully. Plug your M5Atom into your PC and set the flash mode to "DIO" (it defaults to "QIO") and baud rate 115200 - and upload. If you're successful you'll see the M5Atom start to flash. I configured mine through the Serial monitor on the IDE - its now successfully connected to my wifi and waiting to talk to the LIN transcoder circuit and work as a homekit device. Will update this post with next progress.

**Comment by a10kiloham** [comment]
FYI this works fine now in PlatformIO with the following tweak:
https://github.com/a10kiloham/FujiHK/blob/master/platformio.ini

Builds fine. I didn't try the access point via Wifi, but just went into the serial console in Platformio and then set the wifi there. I also reset the setup code then it added fine in Home Assistant after that.

**Comment by wildman199** [comment]
So, is there a working version for HomeKit?

**Comment by FOSV** [comment]
@a10kiloham Did you managed to let the stuff work fine for Homekit? Can you help me?
